core/src/main/scala/kafka/utils/NetworkClientBlockingOps.scala

```diff
@@ -56,8 +56,13 @@ class NetworkClientBlockingOps(val client: NetworkClient) extends AnyVal {
    */
   def blockingReady(node: Node, timeout: Long)(implicit time: JTime): Boolean = {
     require(timeout >=0, "timeout should be >= 0")
-    client.ready(node, time.milliseconds()) || pollUntil(timeout) { (_, now) =>
-      if (client.isReady(node, now))
+
+    // poll once to receive pending disconnects
+    val now = time.milliseconds()
+    client.poll(0, now)
+
+    client.ready(node, now) || pollUntil(timeout) { (_, now) =>
+      if (client.ready(node, now))
```

Comment:
Hmm... I think we may be able to revert this. I was initially concerned that it might be possible to miss the disconnect transition and get stuck in poll with no pending connection. Actually this would be possible if the NetworkClient was configured with a positive reconnect backoff value. If we had just disconnected, the call to ready() would not initiate another connection until the backoff period had expired, which means  there wouldn't be any event pending when we called poll(). Do you think it's OK to assume no backoff in NetworkClient here?